server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1932,8 +1963,10 @@ public synchronized void close() throws IOException {
         toClose.add(injector.getInstance(ClusterService.class));
         toClose.add(() -> stopWatch.stop().start("node_connections_service"));
         toClose.add(injector.getInstance(NodeConnectionsService.class));
-        toClose.add(() -> stopWatch.stop().start("discovery"));
-        toClose.add(injector.getInstance(Discovery.class));
+        if (FeatureFlags.isEnabled(CLUSTERLESS_FLAG) == false) {
+            toClose.add(() -> stopWatch.stop().start("discovery"));
```

Comment:
possible to group all non-clusterless together and put them in the same block?

Reply:
Maybe. My main concern is that order probably matters. 

For example, in the `stop` method, there's a comment saying that we want to stop `Discovery` early to stop responding to pings. We might be able to stop `GatewayService` early too, but I don't know what side-effect that might have.